src/main/java/client/Character.java

```diff
@@ -150,6 +150,8 @@ public class Character extends AbstractCharacterObject {
     private final AtomicInteger gachaexp = new AtomicInteger();
     private final AtomicInteger meso = new AtomicInteger();
     private final AtomicInteger chair = new AtomicInteger(-1);
+    private long totalExpGained = 0;
+    private static final LinkedList<Map<String, Object>> expGainLogQueue = new LinkedList<>();
```

Comment:
There are threadsafe queues in the standard library better suited for this sort of stuff. Have a look at `java.util.concurrent.LinkedBlockingQueue` and its `drainTo` method.